impl/client/src/main/javascript/web/dojo/pentaho/common/PageControl.js

```diff
@@ -30,161 +30,185 @@
+                this.pagePrevDisabled = this.pageNumber === 1;
+                this.pageNextDisabled = this.pageTotal === this.pageNumber;
+                if (this.pagePrevDisabled) {
+                    domClass.add(this.prevPageControl, "pc_pagePrevDisabled");
+                    this.prevPageControl.setAttribute("tabindex", "-1");
+                } else {
+                    domClass.remove(this.prevPageControl, "pc_pagePrevDisabled");
+                    this.prevPageControl.setAttribute("tabindex", "0");
+                }
+                if (this.pageNextDisabled) {
+                    domClass.add(this.nextPageControl, "pc_pageNextDisabled");
+                    this.nextPageControl.setAttribute("tabindex", "-1");
+                } else {
+                    domClass.remove(this.nextPageControl, "pc_pageNextDisabled");
+                    this.nextPageControl.setAttribute("tabindex", "0");
+                }
+            },
+
+            _changePageNumber: function () {
+                if (this.ignorePageNumberUpdate) {
+                    return;
+                }
+                if (!this.pageNumberInput.isValid()) {
+                    this._resetPageNumber();
+                    return;
+                }
+                this.setPageNumber(this.pageNumberInput.get("value"));
+            },
+
+            _updateInternalPageNumber: function (pageNumber) {
+                if (this.pageNumber === pageNumber) {
+                    return;
+                }
+
+                this.ignorePageNumberUpdate = true;
+                try {
+                    if (this.changePageNumberCallback) {
+                        this.changePageNumberCallback(pageNumber);
+                    }
+                    this.pageNumber = pageNumber;
+                    this.pageNumberInput.set("value", this.pageNumber);
+                    this._updatePageButtonState();
+                } finally {
+                    this.ignorePageNumberUpdate = false;
+                }
+                this.pageNumber = pageNumber;
+                this.pageNumberInput.set("value", this.pageNumber);
+                this._updatePageButtonState();
+            },
+
+            _pageNumberKeyUp: function (event) {
+                switch (event.keyCode) {
+                    case keys.ENTER:
+                        this._changePageNumber();
+                        break;
+                    case keys.ESCAPE:
+                        this._resetPageNumber();
+                        break;
+                    default:
+                        // We don't care about any other key
+                        break;
+                }
+            },
+
+            _resetPageNumber: function () {
+                this.ignorePageNumberUpdate = true;
+                try {
+                    this.pageNumberInput.set("value", this.pageNumber);
+                } finally {
+                    this.ignorePageNumberUpdate = false;
+                }
```

Comment:
Why the try/finally block. Is there the expectation for an error to be thrown? Or, just out of extra precaution?

Reply:
I now realize all these comments apply equally to the https://github.com/pentaho/pentaho-platform-plugin-interactive-reporting/blob/wcag/web/src/main/javascript/pir/pir/PageControl.js you based these changes on...